### PR TITLE
Performance Update: Libraries update

### PR DESCRIPTION
**Pull request by oliverfoster**
For Backbone I've created a small compatibility layer for Backbone v1.0.0 to help us migrate.

Handlebars is the only library not updated as grunt-contrib-handlebars does not support handlebars version 3.

swfObject library removed as no longer required.

Release dates and version numbers added to each file.

Included:

Backbone v1.1.2 (plus v1.0.0 compatibility layer) https://backbone.org

Bowser v0.7.2 https://github.com/ded/bowser

Console-Shim Apr 1, 2014 https://github.com/kayahr/console-shim  

Handlebars v2.0.0 (version 3 doesn't work with grunt yet) http://handlebarsjs.com/

jQuery.ImageReady Aug 1, 2013 https://github.com/weblinc/jquery-imageready

jQuery.Inview Jul 10, 2014 https://github.com/protonet/jquery.inview

jQuery.A11y Mar 12, 2015 https://github.com/cgkineo/jquery.a11y

jQuery v1.11.2 https://jquery.org

jQuery v2.1.3 https://jquery.org

json2 Feb 4, 2014 https://cdnjs.cloudflare.com/ajax/libs/json2

Modernizr v2.8.3 http://modernizr.com/download/#-fontface-backgroundsize-borderimage-borderradius-boxshadow-flexbox-flexboxlegacy-hsla-multiplebgs-opacity-rgba-textshadow-cssanimations-csscolumns-generatedcontent-cssgradients-cssreflections-csstransforms-csstransforms3d-csstransitions-applicationcache-canvas-audio-video-input-inputtypes-touch-shiv-cssclasses-teststyles-testprop-testallprops-prefixes-domprefixes-load

RequireJS v2.1.17 http://requirejs.org/

jQuery.ScrollTo v2,1.0 https://github.com/flesler/jquery.scrollTo

Underscore v1.8.3 http://underscorejs.org/

Velocity v1.2.2 http://julian.com/research/velocity/

### CODE REVIEW

**Comment by oliverfoster** [comment]
Now user interaction is working on the views at present. Mouse clicks not binding for some reason. Will investigate.

File: src/core/js/libraries/modernizr.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
...
...d||a.both,j=a.callback||f,k=j,l=a.complete||f,m,n;c(h?a.yep:a.nope,!!i),i&&c(i)}var i,j,l=this.yepnope.loader;if(e(a))g(a,0,l,0);else if(w(a))for(i=0;i<a.length;i++)j=a[i],e(j)?g(j,0,l,0):w(j)?B(j):Object(j)===j&&h(j,l);else Object(a)===a&&h(a,l)},B.addPrefix=function(a,b){z[a]=b},B.addFilter=function(a){x.push(a)},B.errorTimeout=1e4,null==b.readyState&&b.addEventListener&&(b.readyState="loading",b.addEventListener("DOMContentLoaded",A=function(){b.removeEventListener("DOMContentLoaded",A,0),b.readyState="complete"},0)),a.yepnope=k(),a.yepnope.executeStack=h,a.yepnope.injectJs=function(a,c,d,e,i,j){var k=b.createElement("script"),l,o,e=e||B.errorTimeout;k.src=a;for(o in d)k.setAttribute(o,d[o]);c=j?h:c||f,k.onreadystatechange=k.onload=function(){!l&&g(k.readyState)&&(l=1,c(),k.onload=k.onreadystatechange=null)},m(function(){l||(l=1,c(1))},e),i?k.onload():n.parentNode.insertBefore(k,n)},a.yepnope.injectCss=function(a,c,d,e,g,i){var e=b.createElement("link"),j,c=i?h:c||f;e.href=a,e.rel="stylesheet",e.type="text/css";for(j in d)e.setAttribute(j,d[j]);g||(n.parentNode.insertBefore(e,n),m(c,0))}}(this,document),Modernizr.load=function(){yepnope.apply(window,[].slice.call(arguments,0))};
```
Comment (hrajotia) [inline src/core/js/libraries/modernizr.js]: @oliverfoster are we not adding any CSS3, HTML5 specific tests to Modenizr?
We are still using `Modernizr.touch` feature (e.g. [device.js](http://github.com/adaptlearning/adapt_framework/blob/develop/src/core/js/device.js#L17), [accessibility.js](http://github.com/adaptlearning/adapt_framework/blob/develop/src/core/js/accessibility.js#L72), [adapt-contrib-textInput.js](http://github.com/adaptlearning/adapt-contrib-textInput/blob/develop/js/adapt-contrib-textInput.js#L59)).
I also think we should consider including `Modernizr.input.placeholder` to check if browser does support placeholder or not!

what do you think?

Comment (oliverfoster) [inline src/core/js/libraries/modernizr.js]: Did I leave out modernizr touch? :O

Good spot, it looks as though I've not used the correct url. I'll put it back in with the old options. 

Did you want just the input.placeholder?

I think it's a good idea to move forward with form input elements. Placeholder is a big one to implement in both html5 and html4<, pretty small in just html5. Is it worth the energy for such small effect? Does that argument preclude us from including the modernizr option? (Probably not).

File: src/core/js/libraries/backbone.js [hunk 2]
```diff
@@ -1,44 +1,64 @@
...
+	
+
+	//Replacement view instanciator
+	Backbone.View = function(options) {
+		this.cid = _.uniqueId('view');
+		options || (options = {});
+		_.extend(this, _.pick(options, viewOptions));
+		//THE LINE BELOW IS THE FUNCTIONALITY WE USE
+		this.options = _.omit(options, viewOptions);
+		if (Object.keys(this.options).length > 0) {
```
Comment (hrajotia) [inline src/core/js/libraries/backbone.js]: `Object.keys` won't work in IE8.

**Comment by hrajotia** [comment]
+1

**Comment by belal-almassri-spongeuk** [comment]
:+1:

**Comment by zenduo** [comment]
+1

**Comment by belal-almassri-spongeuk** [comment]
:+1:

**Comment by hrajotia** [comment]
+1

**Comment by hrajotia** [comment]
+1

**Comment by brian-learningpool** [comment]
+1

**Comment by moloko** [comment]
+1

**Pull request merged**